### PR TITLE
Develop value estimates ppo

### PR DESCRIPTION
**Pull request by vincentpierre**
Reintegrated value estimates 
Needs to modify the proto file agent_action : 
```
syntax = "proto3";

option csharp_namespace = "MLAgents.CommunicatorObjects";
package communicator_objects;

message AgentActionProto {
    repeated float vector_actions = 1;
    string text_actions = 2;
    repeated float memories = 3;
    float value = 4;
}
```
Get the value estimate in an agent Monobehavior by calling `GetValueEstimate()`

### CODE REVIEW

**Comment by awjuliani** [comment]
This looks great, @vincentpierre. My only question is around what we should do to handle Q-values? The current implementation is specific to `V(s)`, which is fine, but we probably want to avoid continuing to add more and more things to pass back and forth.

**Comment by vincentpierre** [comment]
You mean making value a float list instead of a single float ? We could do that but we would need an environment that has such Q-values.

File: python/unityagents/environment.py [hunk 1]
```diff
@@ -483,8 +495,10 @@ def _generate_step_input(self, vector_action, memory, text_action) -> UnityRLInp
                 action = AgentActionProto(
                     vector_actions=vector_action[b][i*_a_s: (i+1)*_a_s],
                     memories=memory[b][i*_m_s: (i+1)*_m_s],
-                    text_actions=text_action[b][i]
+                    text_actions=text_action[b][i],
```
Comment (awjuliani) [inline python/unityagents/environment.py]: Why is a comma added here?

**Comment by awjuliani** [comment]
@vincentpierre Since we don't have an algorithm which uses this right now, I agree we don't need it. Just gave the PR another pass through, and it looks good to me.

**Pull request merged**